js/source/tile_pyramid.js

```diff
@@ -340,5 +346,52 @@ TilePyramid.prototype = {
                 };
             }
         }
+    },
+
+    /**
+     * For a given tile id find its parent tile from the index
+     * @param {string|number} id tile id
+     * @returns {Object} tile
+     * @private
+     */
+    indexSearch: function (id) {
+        var tile = TileCoord.fromID(id);
+
+        var ids = [];
+
+        var parentTile = tile;
+        while (id !== 0) {
+          parentTile = parentTile.parent();
+          id = parentTile.id;
+          ids.push(id);
+        }
+
+        var cursor = this.index,
+            cursorId = ids.pop(),
+            index;
+
+        var pluckId = function (coord) {
+          return coord.id;
+        };
+
+        while (ids.length) {
+            id = ids.pop();
+            tile = TileCoord.fromID(cursorId);
+            index = tile.children(this.maxzoom).map(pluckId).indexOf(id);
```

Comment:
Is this hot code? Seems like the creation of function, a parallel array, and then another `o(n)` indexOf search would be a potential perf problem.